components/director/internal/domain/fetchrequest/converter_test.go

```diff
@@ -97,3 +101,145 @@ func TestConverter_InputFromGraphQL(t *testing.T) {
 		})
 	}
 }
+
+func TestConverter_FromEntity(t *testing.T) {
+	timestamp := time.Now()
+
+	// given
+	testCases := []struct {
+		Name               string
+		Input              fetchrequest.Entity
+		Expected           model.FetchRequest
+		ExpectedErrMessage string
+	}{
+		{
+			Name:               "All properties given",
+			Input:              fixFetchRequestEntity(t, "1", timestamp),
+			Expected:           fixFetchRequestModel("1", timestamp),
+			ExpectedErrMessage: "",
+		},
+		{
+			Name: "Empty value",
+			Input: fetchrequest.Entity{
+				ID:              "2",
+				TenantID:        "tenant",
+				Auth:            sql.NullString{},
+				StatusTimestamp: timestamp,
+				StatusCondition: string(model.FetchRequestStatusConditionFailed),
+			},
+			Expected: model.FetchRequest{
+				ID:     "2",
+				Tenant: "tenant",
+				Auth:   nil,
+				Status: &model.FetchRequestStatus{
+					Timestamp: timestamp,
+					Condition: model.FetchRequestStatusConditionFailed,
+				},
+			},
+			ExpectedErrMessage: "",
+		},
+		{
+			Name: "Error",
+			Input: fetchrequest.Entity{
+				Auth: sql.NullString{
```

Comment:
Maybe is worth to create function similar to `newNullableString` from `repo` package, but string as parameter.